SitePackage.lua

```diff
@@ -124,38 +125,53 @@ local function startup_hook(usrCmd)
 end
 
 
-local function msg_hook(mode, output)
-    -- mode is avail, list, ...
-    -- output is a table with the current output
+local function errwarnmsg_hook(kind, key, msg, t)
+    -- kind is either lmoderror, lmodwarning or lmodmessage
+    -- key is a unique key for the message (see messageT.lua)
+    -- msg is the actual message to display (as string)
+    -- t is a table with the keys used in msg
+    dbg.start{"errwarnmsg_hook"}
 
-    dbg.start{"msg_hook"}
+    dbg.print{"kind: ", kind}
+    dbg.print{"Msg: ", msg}
+    dbg.print{"keys: ", t}
 
-    dbg.print{"Mode is ", mode, "\n"}
-
-    if output[1] and output[1]:find("Your site prevents the automatic swapping of modules with same name") then
+    if key == "e_No_AutoSwap" then
         -- find the module name causing the issue (almost always toolchain module)
-        local sname  = output[1]:match("$ module swap ([^ ]+)%s+[^ \n]+")
-        local mStack = ModuleStack:moduleStack()
+        local sname = t.sn
+        local frameStk = FrameStk:singleton()
 
         local errmsg = {"A different version of the '"..sname.."' module is already loaded (see output of 'ml')."}
-        if not mStack:empty() then
-            errmsg[#errmsg+1] = "You should load another '"..mStack:sn().."' module for that is compatible with the currently loaded version of '"..sname.."'."
-            errmsg[#errmsg+1] = "Use 'ml spider "..mStack:sn().."' to get an overview of the available versions."
+        if not frameStk:empty() then
+            local compat_msg = "' module for that is compatible with the currently loaded version of '"
```

Comment:
because otherwise the line is too long, and then `luacheck` spits out a warning that makes Jenkins block the PR... :)